Edited the file for readability

cutefluffykittens opened the pull request.

cutefluffykittens committed:
Edited the file for readability

cutefluffykittens committed:
Created test file for GameMaker

cutefluffykittens committed:
Added more tests

cutefluffykittens committed:
Finished early implementation of tests for lab

cutefluffykittens committed:
Fixed unit test modulability

cutefluffykittens committed:
Added test suite

cutefluffykittens committed:
Got rid of check status

functionality changes

cutefluffykittens committed:
Fixed logout tests

cutefluffykittens committed:
Made tests more modular

cutefluffykittens committed:
Added and fixed tests

cutefluffykittens committed:
Added functionality

Logout and Login

cutefluffykittens committed:
Added CheckStatus test

cutefluffykittens committed:
Added checkStatus tests

cutefluffykittens committed:
Implemented checkStatus method

tekkeon commented on gamemaker.py:
```diff
@@ -1,12 +1,24 @@
-      self.loggedin = true
-  //TODO Implement checkStatus
+    returnValue = ""
+    if (self.database.curUser != name && self.database.gameMakerCred = {name, password}):
```
- Remove parentheses after if
- curUser is of type User, and name is a string; change this to self.database.curUser.name
- Change && to "and" (without quotes)

tekkeon commented on gamemaker.py:
```diff
@@ -1,12 +1,24 @@
-  //TODO Implement checkStatus
+    returnValue = ""
+    if (self.database.curUser != name && self.database.gameMakerCred = {name, password}):
+      self.database.curUser = name
```
Same problem as mentioned above

tekkeon commented on gamemaker.py:
```diff
@@ -1,12 +1,24 @@
+    returnValue = ""
+    if (self.database.curUser != name && self.database.gameMakerCred = {name, password}):
+      self.database.curUser = name
+      returnValue = "User " + name + " logged in!"
```
Are we returning this or just printing it? And if you are returning it, you forgot to return it.

tekkeon commented on gamemaker.py:
```diff
@@ -1,12 +1,24 @@
+    elif: (self.database.curUser == name) returnValue = "" + name + " already logged in!"
+    else: returnValue = "Bad username or password!"
+    
   def checkStatus(self):
```
check_status, not checkStatus

tekkeon commented on gamemaker.py:
```diff
@@ -1,12 +1,24 @@
   def checkStatus(self):
-  //TODO Implement logout
-  def logout(name):
+    string = ' '.join(self.database.teams.keys())
```
?

tekkeon commented on tests/ut-gamemaker.py:
```diff
@@ -0,0 +1,69 @@
+    self.maker1 = gamemaker.GameMaker()
+  
+  def test_BadLogin(self):
+     self.assertEquals(self.maker1.login("Notvalidusername","Notvalidpassword"), "Bad username or password!", "Bad response to invalid login")
```
Should be written as:

def test_bad_login(self):
    self.maker1.login()
    assertEquals(self.database.curUser, self.maker1, "Error Message")

Same for tests below

cutefluffykittens committed:
Cleaned up syntax

cutefluffykittens committed:
Fixed names and check_status

tekkeon commented on gamemaker.py:
```diff
@@ -1,12 +1,30 @@
+
+    def login(self, name, password):
+        returnValue = ""
+        if self.database.curUser.name != name and self.database.gameMakerCred == {name, password}:
```
Gonna change this one more time...

if self.database.get_current_user().name

cutefluffykittens committed:
Fixed login

cutefluffykittens committed:
Changed return values

True or False

cutefluffykittens committed:
Updated tests for true/false

tekkeon commented on gamemaker.py:
```diff
@@ -1,12 +1,28 @@
+            if name == "maker" and maker_cred["maker"] == password:
+                self.database.set_current_user(self)
+                returnValue = True
+        return returnValue
```
Need else statements for both if statements

tekkeon commented on gamemaker.py:
```diff
@@ -1,12 +1,28 @@
+        current_user = self.database.get_current_user()
+        maker_cred = self.database.get_game_maker_cred()
+        if current_user == None:
+            if name == "maker" and maker_cred["maker"] == password:
```
changed maker_cred["maker"] == password to **maker_cred["password"] == password**

tekkeon commented on gamemaker.py:
```diff
@@ -1,12 +1,28 @@
+        return returnValue
+
+    def check_status(self):
+        dict = self.database.teams.keys()
```
I would cross this statement out entirely and just say: if x in self.database.teams.keys()

tekkeon commented on gamemaker.py:
```diff
@@ -1,12 +1,28 @@
+
+    def check_status(self):
+        dict = self.database.teams.keys()
+        if None in dict:
```
if dict.size() == 0

or whatever the python equivalent of .size() is

tekkeon commented on gamemaker.py:
```diff
@@ -1,12 +1,28 @@
+        if None in dict:
+            return "No teams!"
+        else:
+            return dict
```
Should return a string with the formatted message of what to print out

tekkeon commented on gamemaker.py:
```diff
@@ -1,12 +1,28 @@
+    def logout(self):
+        returnValue = False
+        current_user = self.database.get_current_user()
+        if "maker" in current_user:
```
First check if current_user is None. Then do an elif checking if current_user.username == "maker"

tekkeon commented on tests/ut-gamemaker.py:
```diff
@@ -0,0 +1,66 @@
+     self.assertEquals(self.database.get_current_user(), {}, "Current user is not null")
+  def test_good_login(self):
+     self.assertEquals(self.maker1.login("maker","password"), True, "Maker should have logged in")
+     self.assertEquals(self.database.get_current_user(), {"maker" : "password"}, "Current user should be maker")
```
Not sure yet on the implementation of this. current_user will generally hold an object of type Team, but obviously it won't hold an object of type GameMaker. My guess is that we'll create some other GM class that just has username and password fields, that way it will still work with out implementation using current_user

tekkeon commented on tests/ut-gamemaker.py:
```diff
@@ -0,0 +1,66 @@
+      self.assertEquals(self.maker1.logout(), True, "Bad response to maker logging out correctly")
+    def test_current_user_is_not_maker(self):
+      self.maker1.login("maker","password")
+      self.maker1.logout("maker")
```
logout function doesn't take in any parameters

tekkeon commented on tests/ut-gamemaker.py:
```diff
@@ -0,0 +1,66 @@
+    self.assertEquals(self.maker1.check_status(), "No teams!", "CheckStatus did not return proper response to empty team list")
+    
+  def test_single_team(self):
+    self.team1 = team.Team("Team1")
```
Team() takes in three parameters: name, username, and password

tekkeon commented on tests/ut-gamemaker.py:
```diff
@@ -0,0 +1,66 @@
+    
+  def test_single_team(self):
+    self.team1 = team.Team("Team1")
+    dict = self.maker1.check_status()
```
check_status should return a string to print out

tekkeon commented on tests/ut-gamemaker.py:
```diff
@@ -0,0 +1,66 @@
+  def test_single_team(self):
+    self.team1 = team.Team("Team1")
+    dict = self.maker1.check_status()
+    self.assertEquals("Team1" in dict, True, "Status incorrect for single team existing")
```
Creating a team by using its constructor does not automatically add the team to the database, and therefore couldn't be returned in check_status. You would have to manually add the team since there is no current function for creating a team (from the gamemaker's perspective)

tekkeon commented on tests/ut-gamemaker.py:
```diff
@@ -0,0 +1,66 @@
+    self.assertEquals("Team1" in dict, True, "Status incorrect for single team existing")
+  
+  def test_multiple_teams(self):
+    self.team1 = team.Team("Team1")
```
Parameters, like I mentioned above

tekkeon commented on tests/ut-gamemaker.py:
```diff
@@ -0,0 +1,66 @@
+  def test_multiple_teams(self):
+    self.team1 = team.Team("Team1")
+    self.team2 = team.Team("team2")
+    dict = self.maker1.check_status()
```
Ditto

tekkeon commented on tests/ut-gamemaker.py:
```diff
@@ -0,0 +1,66 @@
+    self.team1 = team.Team("Team1")
+    self.team2 = team.Team("team2")
+    dict = self.maker1.check_status()
+    self.assertEquals("Team1" in dict, True, "Cannot find first entry in two team list")
```
Ditto

cutefluffykittens committed:
Fixed commented code

cutefluffykittens committed:
Fixed commented code

cutefluffykittens committed:
Fixed team syntax

cutefluffykittens committed:
Fixed team syntax

cutefluffykittens committed:
Fixed syntax and check_status

Worked with newfound wisdom of Pycharm and fixed check_status based on new criteria

Mandabee commented on gamemaker.py:
```diff
@@ -1,12 +1,31 @@
+        current_user = self.database.get_current_user()
+        maker_cred = self.database.get_game_maker_cred()
+        if current_user is None:
+            if maker_cred["maker"] == "maker" and maker_cred["password"] == password:
```
Is this hard coding the maker and password in? Or is this dynamically referencing it?

tekkeon replied:
if maker_cred["username"] == "maker"...

If you look at the database, you can see how this is set up

Mandabee commented on gamemaker.py:
```diff
@@ -1,12 +1,31 @@
+    def check_status(self):
+        string = ''
+        teams = self.database.get_teams()
+        if len(teams) > 0:
```
You do not need this line - if Len(teams) is 0 then the for loop will not execute at all (I think)

Mandabee commented on gamemaker.py:
```diff
@@ -1,12 +1,31 @@
+        teams = self.database.get_teams()
+        if len(teams) > 0:
+            for team in teams:
+                string += team.user_name + ' is at ' + team.landmark + ' with ' + team.penalties + ' penalties\n'
```
username not user_name
Also we don't have landmarks or penalties yet so this might break/crash

Mandabee commented on gamemaker.py:
```diff
@@ -1,12 +1,31 @@
+    def logout(self):
+        return_value = False
+        current_user = self.database.get_current_user()
+        if "maker" in current_user:
```
I would do 
if current_user is "maker"
unless current_user is some sort of list

tekkeon replied:
current_user will be an object of type GameMaker or Team, so we'll need to do:
if current_user is self

Mandabee reviewed (changes requested):
I think these things are not code - breaking but are important

tekkeon commented on gamemaker.py:
```diff
@@ -1,12 +1,31 @@
+    def logout(self):
+        return_value = False
+        current_user = self.database.get_current_user()
+        if "maker" in current_user:
```
if current_user is self

cutefluffykittens committed:
Fixed commented code

cutefluffykittens committed:
Added newline to check_status

tekkeon commented on gamemaker.py:
```diff
@@ -1,12 +1,30 @@
+    def logout(self):
+        return_value = False
+        current_user = self.database.get_current_user()
+        if current_user is "maker":
```
if current_user is self

cutefluffykittens committed:
Fixed current_user conditional
